reframe/core/pipeline.py

```diff
@@ -722,10 +722,14 @@ def __new__(cls, *args, **kwargs):
         try:
             prefix = cls._rfm_custom_prefix
         except AttributeError:
-            if osext.is_interactive():
-                prefix = os.getcwd()
-            else:
-                prefix = os.path.abspath(os.path.dirname(inspect.getfile(cls)))
+            try:
+                prefix = cls._rfm_base_test_prefix
+            except AttributeError:
+                if osext.is_interactive():
+                    prefix = os.getcwd()
+                else:
+                    prefix = os.path.abspath(
+                        os.path.dirname(inspect.getfile(cls)))
```

Comment:
Instead of all these nested `try/except`, I would suggest this, which is more compact:

```python
prefix = None
with contextlib.suppress(AttributeError):
    for p in ['_rfm_base_test_prefix', '_rfm_custom_prefix']:
        prefix = getattr(cls, p)

if prefix is None:
    prefix = cls._set_prefix()
```

Reply:
I feel that this changes the behaviour of `_rfm_custom_prefix`. Rewriting the try/excepts as above would mean that `_rfm_custom_prefix` is no longer a 'master override' kind of thing and it would have the same functionality as `_rfm_base_test_prefix`. I've changed this a bit to use the `is_interactive` check also for the `_rfm_base_test_prefix`.

Reply:
I don't understand why what I propose here does not give the "master override" power to `_rfm_custom_prefix`. It will also be the last prefix to try, so it will override `_rfm_pinned_prefix`.

Reply:
Hmm, you mean in respect with the `is_interactive()`?

Reply:
You are right. I am resolving this. Also, the `try/except` is also more pythonic (that's the reason I had it like that initially), although I don't like it very much, but on the other hand if you go with `hasattr()`, then you type twice the attributes. So I'm leaving it like now.